src/iexamine.cpp

```diff
@@ -3983,6 +3990,12 @@ void smoker_finalize(player &, const tripoint &examp)
         return;
     }
 
+    for( auto &it : items ) {
```

Comment:
Is it equivalent to the next for? Can they be merged? I have open PR that is changing this function too and can add changes there.

Reply:
No, they are not equivalent. This loop must be completed before the next one to apply rot correctly.